mesonbuild/ast/introspection.py

```diff
@@ -355,6 +357,11 @@ def func_build_target(self, node: BaseNode, args: T.List[TYPE_nvar], kwargs: T.D
     def is_subproject(self) -> bool:
         return self.subproject != ''
 
+    def evaluate_codeblock(self, node: CodeBlockNode, start: int = 0, end: T.Optional[int] = None) -> None:
```

Comment:
I think you should override `func_subdir()` instead, `evaluate_codeblock()` is called for every if-block, foreach-block, etc.

Reply:
You are right. My problem was to have access to the subdir meson.build ast. 
I tried something else. I think this is better.